org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/IResourceRelocationStrategy.xtend

```diff
@@ -29,6 +29,29 @@ interface IResourceRelocationStrategy {
 
 	def void applyChange(ResourceRelocationContext context)
 
+	/**
+	 * By overriding this method client implementations may explicitly demand for relying
+	 * on the persisted index during a refactoring. 
+	 * 
+	 * Refactoring operations usually rely on a lively created resource index. However,
+	 * in case of refactorings affecting a large amounts of files live resource indexing
+	 * may lead to heap pollution and resultant performance decreases.
+	 * 
+	 * If a persisted index is supported by the runtime environment and if at least one of the
+	 * registered implementations of {@link IResourceRelocationStrategy} demands for relying
+	 * on the persisted index and if, furthermore, required preconditions are satisfied,
+	 * like no open editor is dirty and the persisted index is up-to-date, the refactoring
+	 * operation will rely entirely on the persisted index for, e.g., determining incoming
+	 * references of affected objects, and skip the creation of a live resource index.
```

Comment:
this is a very long sentence with many "if ... and if..."

Could you may rephrase it more like explaining the effect first and then make a list of conditions (with ul/ol) that must be true?

Reply:
Actually I'm a bit in a hurry, will improve that later.

Reply:
No prob. 2.18 ist still a long time to go. Happy easter :)